dashboard/src/components/Table/TestsTable.tsx

```diff
@@ -25,21 +23,68 @@ import TableStatusFilter from './TableStatusFilter';
 const ITEMS_PER_PAGE = 10;
 
 export interface ITestsTable {
-  treeId: string;
+  testHistory: TestHistory[];
 }
 
-const TestsTable = ({ treeId }: ITestsTable): JSX.Element => {
-  const { testPath, treeInfo, origin, tableFilter } = useSearch({
+const TestsTable = ({ testHistory }: ITestsTable): JSX.Element => {
+  const { tableFilter } = useSearch({
     from: '/tree/$treeId/',
   });
   const navigate = useNavigate({ from: '/tree/$treeId' });
-  const { data, isLoading } = useTreeTest(
-    treeId,
-    testPath,
-    treeInfo.gitBranch ?? '',
-    treeInfo.gitUrl ?? '',
-    origin,
-  );
+
+  const data = useMemo((): TPathTests[] => {
```

Comment:
I think we can have a loading state if we use a web worker here

Reply:
Is it really necessary considering the time frame that we have?